### PR TITLE
[22] Merge BETA_JAVA22 to master #2167

### PR DESCRIPTION
**Pull request by jarthana**
## What it does
Fixes https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2167

## Author checklist

- [ ] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -39,6 +39,10 @@
   <profiles>
     <profile>
       <id>build-individual-bundles</id>
+      <properties>
```
Comment (akurtakov) [inline pom.xml:42]: You should not compare Y-build with I-builds as this is what will happen if you merge this in master.

**Review by akurtakov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (SarikaSinha) [reply pom.xml:42]: Yes Y build change is only required for Beta branch.

**Review by akurtakov — APPROVED** [review]
Looks good from releng POV>

**Comment by jukzi** [comment]
@jarthana the checks had shown an error which now is on master
https://ci.eclipse.org/jdt/job/eclipse.jdt.debug-Github/job/master/63/
```
[INFO] eclipse.jdt.debug 4.32.0-SNAPSHOT .................. SUCCESS [  3.645 s]
[INFO] [bundle] JDI Debug Model 3.21.400-SNAPSHOT ......... SUCCESS [ 54.779 s]
[INFO] org.eclipse.jdt.launching 3.21.200-SNAPSHOT ........ FAILURE [ 10.502 s]
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:33 min
[INFO] Finished at: 2024-03-20T09:39:17Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.7-SNAPSHOT:compile (default-compile) on project org.eclipse.jdt.launching: Compilation failure: Compilation failure: 
Error:  /home/runner/work/eclipse.jdt.debug/eclipse.jdt.debug/org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/EECompilationParticipant.java:[237] 
Error:  	} else if (version.startsWith(JavaCore.VERSION_22)) {
Error:  	                                       ^^^^^^^^^^
Error:  VERSION_22 cannot be resolved or is not a field
```

**Comment by jukzi** [comment]
When jdt.core adds API that another bundle uses the updated bundle version should be used.
i.e. /org.eclipse.jdt.launching/META-INF/MANIFEST.MF should now require org.eclipse.jdt.core 3.38

**Comment by jarthana** [comment]
> When jdt.core adds API that another bundle uses the updated bundle version should be used. i.e. /org.eclipse.jdt.launching/META-INF/MANIFEST.MF should now require org.eclipse.jdt.core 3.38

I see this now - org.eclipse.jdt.core;bundle-version="[3.34.0,4.0.0)",
Why is this not good enough?

**Comment by jukzi** [comment]
It pretends launching would compile and run with 3.34. It does not anymore. Normally you would not recognize the error when just using only latest bundles. But sometimes people (or CI Jobs;-)) try to use old versions.

**Comment by jarthana** [comment]
> It pretends launching would compile and run with 3.34. It does not anymore. Normally you would not recognize the error when just using only latest bundles. But sometimes people (or CI Jobs;-)) try to use old versions.

So, what do you suggest? [3.38.0,4.0.0) like we have in jdt.ui or just "3.38.0"?

**Comment by jukzi** [comment]
javadoc of JavaCore.VERSION_22 states it was added @since 3.38, so "3.38" should be fine.

**Comment by jukzi** [comment]
> And why is this a problem only now? We seem to be having 3.34 for a while now?

I can only guess, that it already was wrong, but did not became visible since nobody tried it- So that it worked by chance

**Comment by SarikaSinha** [comment]
We don't restrict it to a version like 3.38 as JDT Core can increase to 3.39 but no impact on launching. so please keep the range of [3.38.0,4.0.0)

**Pull request merged**